internal/credit/postgres_connector/balance.go

```diff
@@ -217,7 +221,8 @@ func (a *PostgresConnector) getBalance(
 				// TODO: do we want this to be settable in ledger
 				FilterSubject: []string{ledger.Subject},
 			}
-			queryParams.From = &period.From
+			fromMinute := period.From.Truncate(time.Minute)
+			queryParams.From = &fromMinute
```

Comment:
This way we are overfetching the meters, counting even the events in the same minute happened before the period.From